lib/oelite/task.py

```diff
@@ -183,24 +183,17 @@ def colon_split(s):
         self._meta = meta
         return meta
 
-
-    def run(self):
+    def prepare_context(self):
         meta = self.meta()
-        function = meta.get_function(self.name)
-
+        self.function = meta.get_function(self.name)
         self.do_cleandirs()
-        cwd = self.do_dirs() or meta.get("B")
-
-        # Setup stdin, stdout and stderr redirection
-        stdin = open("/dev/null", "r")
-        self.logfn = "%s/%s.%s.log"%(function.tmpdir, self.name, meta.get("DATETIME"))
-        self.logsymlink = "%s/%s.log"%(function.tmpdir, self.name)
+        self.cwd = self.do_dirs() or meta.get("B")
+        self.stdin = open("/dev/null", "r")
+        self.logfn = "%s/%s.%s.log"%(self.function.tmpdir, self.name, meta.get("DATETIME"))
+        self.logsymlink = "%s/%s.log"%(self.function.tmpdir, self.name)
         oelite.util.makedirs(os.path.dirname(self.logfn))
         try:
-            if self.debug:
-                logfile = os.popen("tee %s"%self.logfn, "w")
```

Comment:
As I wrote in the updated summary above, I now do the "cat the log-file when the task is done".